hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,59 @@
+# EVM Transaction Response Codes in Hedera
+
+When executing an EVM transaction to a smart contract using `contractCall` in Hedera and it fails, a response code is present in the transaction receipt to indicate the top-level status.
```

Comment:
Note, the specification is only for contractCall operation. 
For contractCreate operation, there could be more cases like ERROR_DECODING_BYTESTRING or CONTRACT_BYTECODE_EMPTY for example.

The [issue](https://github.com/hashgraph/hedera-services/issues/14207) mentions "EVM Transaction", but It seems that it probably refers just to contractCall. Any thoughts?

Reply:
I think it would be valuable - and meet the intent of the issue too - to include contract create.

It would be desirable but probably not necessary to distinguish codes that could be returned by one or the other or both.

Reply:
EVM Transaction is any transaction that is processed by the EVM on Hedera.
So that's `ContractCreate`, `ContractCall`, `EthereumTransaction` and even `ContractCallLocal`.
We should indeed distinguish since the request was to support clients to customize their code for support.
As such a client that always uses EVM tools that end up submitting an `EthereumTransaction` should only be concerned with codes applicable to that flow.

Reply:
I've factored that in the latest update. Please re-check :)